static/app/views/dashboardsV2/widgetBuilder/widgetBuilder.tsx

```diff
@@ -461,7 +459,41 @@ function WidgetBuilder({
     });
   }
 
+  function handleYAxisChange(newYAxis: QueryFieldValue[]) {
+    const aggregateAliasFieldStrings = newYAxis.map(generateFieldAsString);
+
+    for (const index in state.queries) {
+      const queryIndex = Number(index);
+      const query = state.queries[queryIndex];
+
+      const descending = query.orderby.startsWith('-');
+      const orderbyAggregateAliasField = query.orderby.replace('-', '');
+      const prevAggregateAliasFieldStrings = query.aggregates.map(getAggregateAlias);
+      const newQuery = cloneDeep(query);
+      newQuery.aggregates = aggregateAliasFieldStrings;
+      newQuery.fields = [...newQuery.columns, ...aggregateAliasFieldStrings];
+      if (
+        !aggregateAliasFieldStrings.includes(orderbyAggregateAliasField) &&
+        query.orderby !== ''
+      ) {
+        if (prevAggregateAliasFieldStrings.length === newYAxis.length) {
+          // The Field that was used in orderby has changed. Get the new field.
+          newQuery.orderby = `${descending && '-'}${
+            aggregateAliasFieldStrings[
+              prevAggregateAliasFieldStrings.indexOf(orderbyAggregateAliasField)
+            ]
+          }`;
+        } else {
+          newQuery.orderby = '';
+        }
+      }
+
+      handleQueryChange(queryIndex, newQuery);
+    }
+  }
+
   function handleYAxisOrColumnFieldChange(newFields: QueryFieldValue[]) {
```

Comment:
If we've created our own `handleYAxisChange`, should this become `handleColumnFieldChange`?